File: LevelSelectScene.js

var LevelSelectScene = function () {};

LevelSelectScene.prototype.onDidLoadFromCCB = function () {
    GameDataManager.tableBgIndex = 1;
    GameDataManager.updateGameLevel(GAMELEVEL_EASY);
	this.updateCCBElements();
};

LevelSelectScene.prototype.updateCCBElements = function () {
	this.chooseDifficultyLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_CHOOSE_GAMELEVEL));
    this.gameLevelEasyButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_EASY));
    this.gameLevelNormalButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_NORMAL));
    this.gameLevelHardButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_HARD));
    this.gameLevelEasyInfoLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_EASY_INFO));
    this.gameLevelNormalInfoLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_NORMAL_INFO));
    this.gameLevelHardInfoLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_GAMELEVEL_HARD_INFO));
    this.chooseTableBgLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_CHOOSE_TABLE));
    this.startPlayButtonLabel.setString(MultiLanguageUtil.getLocalizatedStringForKey(STRING_START_GAME));
};

LevelSelectScene.prototype.onPlayPressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    var scene = cc.BuilderReader.loadAsScene("CFSGamePlayScene.ccbi");
    cc.Director.getInstance().replaceScene(cc.TransitionFade.create(1.0, scene));
};

LevelSelectScene.prototype.onEasyModePressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    GameDataManager.updateGameLevel(GAMELEVEL_EASY);
    this.gameLevelEasyMenuItem.setEnabled(false);
    this.gameLevelNormalMenuItem.setEnabled(true);
    this.gameLevelHardMenuItem.setEnabled(true);
};

LevelSelectScene.prototype.onNormalModePressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    GameDataManager.updateGameLevel(GAMELEVEL_NORMAL);
    this.gameLevelEasyMenuItem.setEnabled(true);
	this.gameLevelNormalMenuItem.setEnabled(false);
	this.gameLevelHardMenuItem.setEnabled(true);
};

LevelSelectScene.prototype.onHardModePressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    GameDataManager.updateGameLevel(GAMELEVEL_HARD);
    this.gameLevelEasyMenuItem.setEnabled(true);
	this.gameLevelNormalMenuItem.setEnabled(true);
	this.gameLevelHardMenuItem.setEnabled(false);
};

LevelSelectScene.prototype.onLeftArrowPressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    if (GameDataManager.tableBgIndex == 1) return;
    this.selectTableBgSprite.setTexture(cc.TextureCache.getInstance().addImage("tableBg{0}.jpg".format(--GameDataManager.tableBgIndex)));
};

LevelSelectScene.prototype.onRightArrowPressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    if (GameDataManager.tableBgIndex == 5) return;
    this.selectTableBgSprite.setTexture(cc.TextureCache.getInstance().addImage("tableBg{0}.jpg".format(++GameDataManager.tableBgIndex)));
};

LevelSelectScene.prototype.onHomePressed = function () {
	if (GameDataManager.isSoundEffectOn) cc.AudioEngine.getInstance().playEffect("menu.mp3");
    var scene = cc.BuilderReader.loadAsScene("MainMenuScene.ccbi");
    cc.Director.getInstance().replaceScene(cc.TransitionFade.create(1.0, scene));
};

File: ClearFiveStages-iOS/ClearFiveStages/Scene/LevelSelectScene.m
#import "LevelSelectScene.h"
#import "GameDataManager.h"
#import "SimpleAudioEngine.h"
#import "MultiLanguageUtil.h"
#import "CCBReader.h"

@implementation LevelSelectScene

-(void)didLoadFromCCB {
	// update elements for multi-language support
    [[GameDataManager sharedManager] updateGameLevel:GAMELEVEL_EASY];
	[GameDataManager sharedManager].tableBgIndex = 1;
	[self updateCCBElements];
}

-(void) onEnterTransitionDidFinish {}

// multi-language handle
-(void) updateCCBElements {
    [chooseDifficultyLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_CHOOSE_GAMELEVEL]];
    [gameLevelEasyButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_EASY]];
    [gameLevelNormalButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_NORMAL]];
    [gameLevelHardButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_HARD]];
    [gameLevelEasyInfoLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_EASY_INFO]];
    [gameLevelNormalInfoLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_NORMAL_INFO]];
    [gameLevelHardInfoLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_GAMELEVEL_HARD_INFO]];
    [chooseTableBgLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_CHOOSE_TABLE]];
    [startPlayButtonLabel setString:[MultiLanguageUtil getLocalizatedStringForKey:STRING_START_GAME]];
}

-(void)onPlayPressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
    CCScene* scene = [CCBReader sceneWithNodeGraphFromFile:@"CFSGamePlayScene.ccbi"];
	[[CCDirector sharedDirector] replaceScene:[CCTransitionFadeBL transitionWithDuration:1.0f scene:scene]];
}

-(void)onLeftArrowPressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	if ([GameDataManager sharedManager].tableBgIndex == 1) return;
	[selectTableBgSprite setTexture:[[CCTextureCache sharedTextureCache] addImage:[NSString stringWithFormat:@"tableBg%d.jpg", --[GameDataManager sharedManager].tableBgIndex]]];
}

-(void)onRightArrowPressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	if ([GameDataManager sharedManager].tableBgIndex == 5) return;
	[selectTableBgSprite setTexture:[[CCTextureCache sharedTextureCache] addImage:[NSString stringWithFormat:@"tableBg%d.jpg", ++[GameDataManager sharedManager].tableBgIndex]]];
}

-(void)onEasyModePressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	[[GameDataManager sharedManager] updateGameLevel:GAMELEVEL_EASY];
	[gameLevelEasyMenuItem setIsEnabled:NO];
	[gameLevelNormalMenuItem setIsEnabled:YES];
	[gameLevelHardMenuItem setIsEnabled:YES];
}

-(void)onNormalModePressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	[[GameDataManager sharedManager] updateGameLevel:GAMELEVEL_NORMAL];
	[gameLevelEasyMenuItem setIsEnabled:YES];
	[gameLevelNormalMenuItem setIsEnabled:NO];
	[gameLevelHardMenuItem setIsEnabled:YES];
}

-(void)onHardModePressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	[[GameDataManager sharedManager] updateGameLevel:GAMELEVEL_HARD];
	[gameLevelEasyMenuItem setIsEnabled:YES];
	[gameLevelNormalMenuItem setIsEnabled:YES];
	[gameLevelHardMenuItem setIsEnabled:NO];
}

-(void) onHomePressed {
	if ([GameDataManager sharedManager].isSoundEffectOn == YES) [[SimpleAudioEngine sharedEngine] playEffect:@"menu.mp3"];
	CCScene* scene = [CCBReader sceneWithNodeGraphFromFile:@"MainMenuScene.ccbi"];
	[[CCDirector sharedDirector] replaceScene:[CCTransitionFadeBL transitionWithDuration:1.0f scene:scene]];
}

@end
